pkg/option/option.go

```diff
@@ -115,200 +120,224 @@ func (om OptionMap) DeepCopy() OptionMap {
+func (io *IntOptions) GetImmutableModel() *models.ConfigurationMap {
 	immutableCfg := make(models.ConfigurationMap)
 	return &immutableCfg
 }
 
 // GetMutableModel returns the set of mutable options as a ConfigurationMap API model.
-func (bo *BoolOptions) GetMutableModel() *models.ConfigurationMap {
+func (io *IntOptions) GetMutableModel() *models.ConfigurationMap {
 	mutableCfg := make(models.ConfigurationMap)
-	bo.optsMU.RLock()
-	for k, v := range bo.Opts {
-		if v {
-			mutableCfg[k] = "Enabled"
+	io.optsMU.RLock()
+	for k, v := range io.Opts {
+		if v == OptionDisabled {
+			mutableCfg[k] = fmt.Sprintf("Disabled")
 		} else {
-			mutableCfg[k] = "Disabled"
+			mutableCfg[k] = fmt.Sprintf("Enabled")
 		}
 	}
-	bo.optsMU.RUnlock()
+	io.optsMU.RUnlock()
 
 	return &mutableCfg
 }
 
-func (bo *BoolOptions) DeepCopy() *BoolOptions {
-	bo.optsMU.RLock()
-	cpy := &BoolOptions{
-		Opts:    bo.Opts.DeepCopy(),
-		Library: bo.Library,
+func (io *IntOptions) DeepCopy() *IntOptions {
+	io.optsMU.RLock()
+	cpy := &IntOptions{
+		Opts:    io.Opts.DeepCopy(),
+		Library: io.Library,
 	}
-	bo.optsMU.RUnlock()
+	io.optsMU.RUnlock()
 	return cpy
 }
 
-func NewBoolOptions(lib *OptionLibrary) *BoolOptions {
-	return &BoolOptions{
+func NewIntOptions(lib *OptionLibrary) *IntOptions {
+	return &IntOptions{
 		Opts:    OptionMap{},
 		Library: lib,
 	}
 }
 
-func (bo *BoolOptions) isEnabled(key string) bool {
-	set, exists := bo.Opts[key]
-	return exists && set
+func (io *IntOptions) GetValue(key string) int {
+	value, exists := io.Opts[key]
+	if !exists {
+		return OptionDisabled
+	}
+	return value
+}
+
+func (io *IntOptions) IsEnabled(key string) bool {
+	io.optsMU.RLock()
+	defer io.optsMU.RUnlock()
+	return io.GetValue(key) != OptionDisabled
 }
 
-func (bo *BoolOptions) IsEnabled(key string) bool {
-	bo.optsMU.RLock()
-	defer bo.optsMU.RUnlock()
-	return bo.isEnabled(key)
+// SetValidated sets the option `key` to the specified value. The caller is
+// expected to have validated the input to this function.
+func (io *IntOptions) SetValidated(key string, value int) {
+	io.optsMU.Lock()
+	io.Opts[key] = value
+	io.optsMU.Unlock()
 }
 
-func (bo *BoolOptions) Set(key string, value bool) {
-	bo.optsMU.Lock()
-	bo.Opts[key] = value
-	bo.optsMU.Unlock()
+// SetBool sets the specified option to Enabled.
+func (io *IntOptions) SetBool(key string, value bool) {
+	intValue := OptionDisabled
+	if value {
+		intValue = OptionEnabled
+	}
+	io.optsMU.Lock()
+	io.Opts[key] = intValue
+	io.optsMU.Unlock()
 }
 
-func (bo *BoolOptions) Delete(key string) {
-	bo.optsMU.Lock()
-	delete(bo.Opts, key)
-	bo.optsMU.Unlock()
+func (io *IntOptions) Delete(key string) {
+	io.optsMU.Lock()
+	delete(io.Opts, key)
+	io.optsMU.Unlock()
 }
 
-func (bo *BoolOptions) SetIfUnset(key string, value bool) {
-	bo.optsMU.Lock()
-	if _, exists := bo.Opts[key]; !exists {
-		bo.Opts[key] = value
+func (io *IntOptions) SetIfUnset(key string, value int) {
```

Comment:
This is extremely nit-picky, but I dislike the use of 'io' as a receiver name because it may look like we are using https://golang.org/pkg/io/ instead of int options.